### PR TITLE
Fix "Stateful set integration when Single CQ should admit group that fits" test.

### LINKED ISSUE #3299: [Flaky e2e test]  Stateful set integration when Single CQ should admit group that fits
/kind flake 

**What happened**:

The test flaked https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/periodic-kueue-test-e2e-main-1-31/1849350043572637696

**What you expected to happen**:

no flakes

**How to reproduce it (as minimally and precisely as possible)**:

run on CI

**Anything else we need to know?**:

```
End To End Suite: kindest/node:v1.31.0: [It] Stateful set integration when Single CQ should admit group that fits expand_less	18s
{Timed out after 5.001s.
The matcher passed to Eventually returned the following error:
    <*errors.errorString | 0xc000212480>: 
    NotFoundError expects an error
    {
        s: "NotFoundError expects an error",
    } failed [FAILED] Timed out after 5.001s.
The matcher passed to Eventually returned the following error:
    <*errors.errorString | 0xc000212480>: 
    NotFoundError expects an error
    {
        s: "NotFoundError expects an error",
    }
In [It] at: /home/prow/go/src/kubernetes-sigs/kueue/test/e2e/singlecluster/statefulset_test.go:130 @ 10/24/24 07:28:30.802
}
```

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind bug
/kind failing-test
/kind flake

#### What this PR does / why we need it:
Fix "Stateful set integration when Single CQ should admit group that fits" test.

#### Which issue(s) this PR fixes:

Fixes #3299

#### Special notes for your reviewer:
Removing a StatefulSet sometimes takes more than 5 seconds. However, we don't need to wait for the StatefulSet to be completely removed; instead, we just need to ensure that the workload is removed.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/singlecluster/statefulset_test.go [hunk 1]
```diff
@@ -126,11 +126,11 @@ var _ = ginkgo.Describe("Stateful set integration", func() {
 				conflictingWorkload := &kueue.Workload{}
 				gomega.Expect(k8sClient.Get(ctx, conflictingWlLookupKey, conflictingWorkload)).To(gomega.Succeed())
 				gomega.Expect(createdWorkload.Name).ToNot(gomega.Equal(conflictingWorkload.Name))
-				util.ExpectObjectToBeDeleted(ctx, k8sClient, conflictingStatefulSet, true)
```
Comment (mimowo) [inline test/e2e/singlecluster/statefulset_test.go:129]: if it works but takes more time can we have a new function `util.ExpectObjectToBeDeletedWithTimeout` and pass the LongTimeout instead

Comment (mbobrovskyi) [reply test/e2e/singlecluster/statefulset_test.go:129]: Done

**Comment by mszadkow** [comment]
now it's even better ;)
/lgtm

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve
Thanks!

**Pull request merged**